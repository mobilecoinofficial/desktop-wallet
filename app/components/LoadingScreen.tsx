import React, { useEffect } from 'react';
import type { FC } from 'react';

import { Box, LinearProgress, makeStyles } from '@material-ui/core';
import NProgress from 'nprogress';

import type { Theme } from '../theme';

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    height: '100%',
    justifyContent: 'center',
    minHeight: '100%',
    padding: theme.spacing(3),
  },
}));

const LoadingScreen: FC = () => {
  const classes = useStyles();

  useEffect(() => {
    NProgress.start();

    return () => {
      NProgress.done();
    };
  }, []);

  return (
    <div className={classes.root}>
      <Box width={400}>
        <LinearProgress />
      </Box>
    </div>
  );
};

export default LoadingScreen;
