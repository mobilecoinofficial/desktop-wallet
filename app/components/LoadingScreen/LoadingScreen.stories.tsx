import React, { ComponentProps } from 'react';

import { Story } from '@storybook/react';

import LoadingScreen from './LoadingScreen.component';

export default {
  component: LoadingScreen,
  title: 'Components/LoadingScreen',
};

const Template: Story<ComponentProps<typeof LoadingScreen>> = (args) => <LoadingScreen {...args} />;

export const Primary = Template.bind({});
Primary.args = {};
