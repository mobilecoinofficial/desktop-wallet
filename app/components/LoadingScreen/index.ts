import { LoadingScreen } from './LoadingScreen.view';

export default LoadingScreen;
export { LoadingScreen };
