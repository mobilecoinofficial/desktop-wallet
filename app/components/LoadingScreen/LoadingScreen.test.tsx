import React from 'react';

import { render } from '@testing-library/react';

import { LoadingScreen } from './LoadingScreen.view';

describe('LoadingScreen', () => {
  test('renders to screen when called', () => {
    const { asFragment } = render(<LoadingScreen />);
    expect(asFragment()).toMatchSnapshot();
  });
});
