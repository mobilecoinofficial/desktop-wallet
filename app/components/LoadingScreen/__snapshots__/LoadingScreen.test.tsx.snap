// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen renders to screen when called 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <div
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
        role="progressbar"
      >
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
        />
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
